### PR TITLE
Update all Azure.Identity references to 1.10.2, fix related examples

### PR DESCRIPTION
**Pull request by amerjusupovic**

### CODE REVIEW

File: examples/DotNetCore/WebDemo/Microsoft.Azure.AppConfiguration.WebDemo/Microsoft.Azure.AppConfiguration.WebDemo.csproj [hunk 1]
```diff
@@ -1,10 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.0</TargetFramework>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (avanigupta) [inline examples/DotNetCore/WebDemo/Microsoft.Azure.AppConfiguration.WebDemo/Microsoft.Azure.AppConfiguration.WebDemo.csproj:3]: net5.0 is already out of support. Lets go with net6.0.

**Comment by avanigupta** [comment]
One more reference needs to be updated: https://github.com/Azure/AppConfiguration/blob/ba507988685cd2a2811ce4d48515367b9bb76acf/examples/DotNetFramework/WebDemo/Microsoft.Azure.AppConfiguration.WebDemo/Microsoft.Azure.AppConfiguration.WebDemo.csproj#L54C20-L54C20

File: examples/DotNetCore/WebDemo/Microsoft.Azure.AppConfiguration.WebDemo/Microsoft.Azure.AppConfiguration.WebDemo.csproj (hunk 1 above)
Comment (amerjusupovic) [reply examples/DotNetCore/WebDemo/Microsoft.Azure.AppConfiguration.WebDemo/Microsoft.Azure.AppConfiguration.WebDemo.csproj:3]: I was thinking the same, but there's already a WebDemoNet6 so it seemed odd to have both

Comment (avanigupta) [reply examples/DotNetCore/WebDemo/Microsoft.Azure.AppConfiguration.WebDemo/Microsoft.Azure.AppConfiguration.WebDemo.csproj]: Are there any major differences between the two? We might not need both samples. I think this one can be deleted in favor of the .NET 6 sample.

Comment (zhenlan) [reply examples/DotNetCore/WebDemo/Microsoft.Azure.AppConfiguration.WebDemo/Microsoft.Azure.AppConfiguration.WebDemo.csproj]: Yes, we can delete this. If we delete this, please update the readme too
https://github.com/Azure/AppConfiguration/blob/main/examples/README.md

**Review by avanigupta — APPROVED** [review]

**Review by zhenlan — APPROVED** [review]

**Pull request merged**